src/lib/pdfTools.ts

```diff
@@ -115,47 +110,43 @@ export async function extractTablesFromJson(
   pdf: Buffer,
   json: ParsedDocument,
   outputDir: string,
-  searchTerm: string
+  searchTerms: string[]
 ): Promise<TableWithFilename[]> {
   const pngs = await getPngsFromPdfPage(pdf)
-  const pages = findRelevantTablesGroupdOnPages(json, searchTerm)
-
-  const tablePromises = pages.flatMap(
-    ({ pageIndex, tables, pageWidth, pageHeight }) =>
-      tables.map((table) =>
-        pngs.getPage(pageIndex + 1).then((png) => {
-          /* Denna fungerar inte än pga boundingbox är fel pga en bugg i NLM ingestor BBOX (se issue här: https://github.com/nlmatics/nlm-ingestor/issues/66). 
-             När den är fixad kan denna användas istället för att beskära hela sidan. */
-          /* TODO: fixa boundingbox för tabeller
-          const { x, y, width, height } = calculateBoundingBoxForTable(
-            table,
-            pageWidth,
-            pageHeight
-          )*/
-          const name = table.name
-            .replace(/[^a-z0-9]/gi, '_')
-            .toLowerCase()
-            .substring(0, MAX_LENGTH_TABLE_NAME)
-          const pngName = `table-${pageIndex}-${name}.png`
-          const filename = path.join(outputDir, pngName)
-
-          const pageWidth2 = Math.floor(pageWidth * 2)
-          const pageHeight2 = Math.floor(pageHeight * 2)
-
-          console.log('extracting screenshot to outputPath', filename)
-          return extractRegionAsPng(
-            png,
-            filename,
-            0,
-            0,
-            pageWidth2,
-            pageHeight2
-          ).then(() => {
-            return { ...table, filename } as TableWithFilename
-          })
-        })
-      )
+  const pages = Object.values(
+    findRelevantTablesGroupedOnPages(json, searchTerms)
+  )
+
+  const processedPages = new Set<number>()
+
+  const tablePromises = pages.map(
+    ({ pageIndex, tables, pageWidth, pageHeight }) => {
+      if (processedPages.has(pageIndex)) {
+        return tables.map((table) =>
+          Promise.resolve({ ...table, filename: null })
+        )
```

Comment:
Hmm wasn't sure how to best handle that scenario. It currently means no screenshot was taken for that table since the page containing that table has already been processed!